swiftide/src/indexing/node.rs

```diff
@@ -39,10 +40,12 @@ pub struct Node {
     pub path: PathBuf,
     /// Data chunk contained in the node.
     pub chunk: String,
-    /// Optional vector representation of the data chunk.
-    pub vector: Option<Vec<f32>>,
+    /// Optional vector representation of embedded data.
+    pub vectors: Option<HashMap<EmbeddableType, Vec<f32>>>,
     /// Metadata associated with the node.
-    pub metadata: HashMap<String, String>,
+    pub metadata: BTreeMap<String, String>,
```

Comment:
I change it to `BTreeMap` to avoid random order in embeddables combining metadata with chunk, and to make testing a bit easier.

Reply:
A disclaimer comment.
Resolving it to calm down PR merging rules.